src/DeviceInterface.cpp

```diff
@@ -46,6 +48,56 @@ bool lookup_runtime_routine(const char *name, const Target &target,
 
 }
 
+namespace Halide {
+halide_device_interface_t *get_default_device_interface_for_target(const Target &t) {
+    return get_device_interface_for_device_api(DeviceAPI::Default_GPU, t);
+}
+
+halide_device_interface_t *get_device_interface_for_device_api(const DeviceAPI &d, const Target &t) {
```

Comment:
I wonder if this function should perhaps not take a target, and instead used get_host_target().with_feature(Target::x) for lookup_runtime_routine, where x depends on the value of d.